backend/app/infrastructure/external_api/etherscan/req_params.py

```diff
@@ -15,3 +17,7 @@ class BlockAction(StrEnum):
 class ClosestValue(StrEnum):
     BEFORE = "before"
     AFTER = "after"
+
+
+def obfuscate_url(api_url: str) -> str:
+    return api_url.replace(app.config["ETHERSCAN_API_KEY"], "<API_KEY>")
```

Comment:
Etherscan api key is just one example of the possible secrets leakage, but there might be more, eg. api keys from other providers we use. I think we should never re-throw error messages from external APIs and return some meaningful message to the client instead. External API messages are not interesting for the client anyway

Reply:
Sure, I stayed with putting obfuscated API URL (we need obfuscation for Etherscan since this is an only API where we put the api key in the url directly, the rest uses headers) but I got rid of putting exception details in the description in backend/app/exceptions.py, please recheck. :v:

Reply:
Even that we currently don't have any other API with a key in the path I'd still change the description in `ExternalApiException` to be sth like: `The request to the external service failed.`. We can easily forget about it when adding some new external API calls with a key in the path, or leak some other sensitive information there. It'll be safer not to include any information about the external service at all

Reply:
changed